tests/lib/prepare.sh

```diff
@@ -774,37 +774,32 @@ EOF
         cd repacked-kernel
         rm -rf firmware/*
 
-        # the code below drops the modules that are not loaded on the
-        # current host, this should work for most cases, since the image will be
-        # running on the same host
-        # TODO:UC20: enable when ready
-
-        # To avoid shellcheck unused code warning, we cannot use "exit 0" to disable
-        # the module drop code. To avoid commented out code, we use a flag instead.
-        # Strip off the check when this UC20 code is enabled.
-        uc20Ready=false
-
-        if [ "$uc20Ready" = "true" ]; then
-            # drop unnecessary modules
-            awk '{print $1}' <  /proc/modules  | sort > /tmp/mods
-            #shellcheck disable=SC2044
-            for m in $(find modules/ -name '*.ko'); do
-                noko=$(basename "$m"); noko="${noko%.ko}"
-                if echo "$noko" | grep -f /tmp/mods -q ; then
-                    echo "keeping $m - $noko"
-                else
-                    rm -f "$m"
-                fi
-            done
-            #shellcheck disable=SC2010
-            kver=$(ls "config"-* | grep -Po 'config-\K.*')
-
-            # depmod assumes that /lib/modules/$kver is under basepath
-            mkdir -p fake/lib
-            ln -s "$PWD/modules" fake/lib/modules
-            depmod -b "$PWD/fake" -A -v "$kver"
-            rm -rf fake
-        fi
+        # The code below drops the unnecessary modules that are not loaded on the
+        # current host, this works for most cases, since the image will be
+        # running on the same host, however some additional modules are required
+
+        # Load additional required modules on current host
+        modprobe brd br_netfilter arp_tables ip_vs_rr ip_vs_sh ip_vs_wrr overlay libcrc32c
+
+        # Get list of loaded modules from current host
+        awk '{print $1}' <  /proc/modules  | sort > /tmp/mods
```

Comment:
Slightly silly question, would just doing:
```
awk '{print $1}' <  /proc/modules  | sort > /tmp/mods
echo "brd br_netfilter arp_tables ip_vs_rr ip_vs_sh ip_vs_wrr overlay libcrc32c" >> /tmp/mods 
```
also work? Might be simpler then to modprobe. It mostly the same of feels very nitpicky (sorry!).

One small difference is that spread will distribute tests accross machines so not all need e.g. the overlay module, only the machine that actually runs the relevant test.

Reply:
Dropped modprobe approach and adding directly to /tmp/mods, with one entry per line as for `grep -f` to work.

Reply:
In hindsight, the benefit of mod probe is that it loads dependencies as well, but the downside is that the list is then not definitive of which additional modules are added. Had to now manually work out the additional dependencies which caused failures, and now we have definitive list which is nice.